src/linear_algebra/general_linear_group.lean

```diff
@@ -0,0 +1,263 @@
+/-
+
+The Genera Linear group $GL(n, R)$
+-/
+import linear_algebra.matrix
+import linear_algebra.matrix.nonsingular_inverse
+import linear_algebra.special_linear_group
+import linear_algebra.determinant
+
+
+/-!
+# The General Linear group $GL(n, R)$
+
+This file defines the elements of the General Linear group `general_linear_group n R`,
+consisting of all invertible `n` by `n` `R`-matrices.
+
+
+## Main definitions
+
+ * `matrix.GL` is the type of matrices over R with unit determinant
+ * `matrix.GL.group` gives the group structure (under multiplication)
+ * `matrix.GL_plus.GL_pos` gives the subgroup of matrices with
+ positive determinant (over a linear ordered ring)
+
+## Implementation notes
+
+
+## References
+
+
+## Tags
+
+matrix group, group, matrix inverse
+-/
+
+namespace matrix
+universes u v
+open_locale matrix
+open linear_map
+
+
+section
+
+variables (n : Type u) [decidable_eq n] [fintype n] (R : Type v) [comm_ring R]
+
+/-- `GL n R` is the group of `n` by `n` `R`-matrices with unit determinant.
+Defined as a subtype of matrices
+-/
+abbreviation GL (R : Type*) [comm_ring R] :  Type* := units (matrix n n R )
```

Comment:
I know it's a mouthful, but for consistency with `linear_map.general_linear_group` this should probably be called `matrix.general_linear_group`

Reply:
Done. I also had to change the namespace name. I made it `general_linear_group` as well.